src/action.js

```diff
@@ -0,0 +1,182 @@
+import * as os from "os";
+
+import * as github from "@actions/github";
+import * as toolCache from "@actions/tool-cache";
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+
+// Define consts
+
+export const gruntworkOrg = "gruntwork-io";
+export const patcherRepo = "patcher-cli";
+export const patcherVersion = "v0.4.3";
+
+export const reportCommand = "report";
+export const updateCommand = "update";
+
+export const nonInteractiveFlag = "--non-interactive"
+export const noColorFlag = "--no-color"
+export const skipContainerFlag = "--skip-container-runtime"
+
+function osPlatform() {
+    switch (os.platform()) {
+        case "linux":
+            return "linux";
+        case "darwin":
+            return "darwin";
+        default:
+            core.setFailed("Unsupported operating system - the Patcher action is only released for Darwin and Linux");
+            return;
+    }
+}
+
+export async function openPullRequest(output, dependency, ghToken) {
+    const head = `patcher-updates-${dependency}`
+    const title = `[Patcher] Update ${dependency}`
+    const commitMessage = "Update dependencies using Patcher"
+    const commitAuthor = "Grunty"
+    const commitEmail = "grunty@gruntwork.io"
+
+    const body = `
+Updated the \`${dependency}\` dependency using Patcher.
+
+### Update summary
+\`\`\`yaml
+${output}
+\`\`\`
+`
+
+    await exec.exec("git", ["config", "user.name", commitAuthor])
+    await exec.exec("git", ["config", "user.email", commitEmail])
+    await exec.exec("git", ["add", "."])
+    await exec.exec("git", ["commit", "-m", commitMessage])
+    await exec.exec("git", ["checkout", "-b", head])
+
+    const context = github.context;
+
+    await exec.exec("git", ["push", "-f", `https://${ghToken}@github.com/${context.repo.owner}/${context.repo.repo}.git`])
+
+    const octokit = new github.getOctokit(ghToken);
+
+    const repoDetails = await octokit.rest.repos.get({ ...context.repo });
+    const base = repoDetails.data.default_branch;
+    core.debug(`Base branch is ${base}. Opening the PR against it.`)
+
+    try {
+        await octokit.rest.pulls.create({ ...context.repo, title, head, base, body, });
+    } catch (error) {
+        if (error.message?.includes(`A pull request already exists for`)) {
+            core.error(`A pull request for ${head} already exists. The branch was updated.`)
+        } else {
+            throw error
+        }
+    }
+}
+
+export async function downloadRelease(owner, repo, tag, ghToken) {
```

Comment:
I'm not sure about how GitHub handles these actions, but I notice that some of your functions have the `export` keyword and others don't. Is this intentional? It seems like this action is all contained in one file and that no other modules are trying to `import` the functions that you are `exporting` here.

Reply:
I would consider a few other changes to this method:
- I would take `octokit` as a param into this function. This will allow you to provide a `mock` one if you want to test this function later
- I would take `platform` as a param whose type is one of the `allowedPlatforms`..which you could also use in your `isSupportedPlatform` method.

Reply:
Should this function take the `assetName` or `assetNameRegex` as a param given that it's a generic "downloadRelease" function? 

Or should we go in the other direction and make it more specific like: `downloadPatcherBinary` - in which case you might only need the `tag` and `ghToken` params

Reply:
>I'm not sure about how GitHub handles these actions, but I notice that some of your functions have the export keyword and others don't. Is this intentional? It seems like this action is all contained in one file and that no other modules are trying to import the functions that you are exporting here.

This was a leftover from initially separating the functions in files. The constants were also being exported! Fixed.